app/Repositories/Daemon/BaseRepository.php

```diff
@@ -82,6 +82,10 @@ public function setServer(Server $server)
     {
         $this->server = $server;
 
+        if ($this->getNode()->id !== $server->node_id) {
```

Comment:
This should be happening in the `getHttpClient` function I think, otherwise you can still chain this backwards and defeat the purpose of this check.

Reply:
Also can you make sure to get a test case added that catches this.